src/coreclr/jit/lower.cpp

```diff
@@ -1795,7 +1795,9 @@ GenTree* Lowering::AddrGen(void* addr)
 GenTree* Lowering::LowerCallMemmove(GenTreeCall* call)
 {
     assert(comp->lookupNamedIntrinsic(call->gtCallMethHnd) == NI_System_Buffer_Memmove);
-    assert(call->gtArgs.CountArgs() == 3);
+
+    // R2R adds r2r cell arg on arm64
+    assert(comp->opts.IsReadyToRun() || (call->gtArgs.CountArgs() == 3));
```

Comment:
Isn't the R2R cell arg added as one of the first args so that all the arg indices below are shifted in R2R?

You may want to add `GetUserArgByIndex` that skips `IsArgAddedLate()` args.

I also didn't realize this in the earlier PR but it seems like you need to add some handling for (explicit) tailcalls; in particular in the call stubs (with `compHasNextCallRetAddr`) the kind of unconditional removal of the call that happens here would cause problems.

Reply:
Actually, given that `Buffer.Memmove` is internal things are _maybe_ ok. I still think we might hit issue in jitstress-random when Buffer.Memmove is converted to an explicit tailcall, but let's see.